### ISSUE TITLE
weechat does not assign expected prefixes to a user upon joining a channel

### ISSUE THREAD

**Issue by meingtsla** [weechat/weechat#1022]
Version affected: WeeChat 1.9-rc1 (git: v1.9-rc1-6-gfadee5066) [compiled on Jun 19 2017 19:09:10]

Steps to reproduce, after starting up a newly installed weechat from git:

* /server add freenode chat.freenode.net/6697 -ssl
* /connect freenode
* alt+j alt+r (to open irc_raw buffer)
* /join ##some-empty-channel

Observed behavior:
* The irc_raw buffer shows the following: (sanitized output)

```
19:21:46 --> freenode | :nick!~user@host JOIN ##some-empty-channel
19:21:46 --> freenode | :moon.freenode.net MODE ##some-empty-channel +ns
19:21:46 --> freenode | :moon.freenode.net 353 nick @ ##some-empty-channel :@nick
19:21:46 --> freenode | :moon.freenode.net 366 nick ##some-empty-channel :End of /NAMES list.
19:21:48 <-- freenode | MODE ##some-empty-channel
19:21:48 --> freenode | :moon.freenode.net 324 nick ##some-empty-channel +ns
19:21:48 --> freenode | :moon.freenode.net 329 nick ##some-empty-channel 1497925274
```

* However, in the buffer for ##some-empty-channel, the nicklist bar does not show my nickname as having a @ prefix, despite the 353 output clearly showing a @ prefix for me.
* In addition, upon issuing the command /names ##some-empty-channel, the raw 353 output is as seen before, but the observed formatted output is the following, with no prefix.  This is clearly inconsistent with the raw output:
```
19:26:09  -- | Nicks ##some-empty-channel: [nick]
19:26:09  -- | Channel ##some-empty-channel: 1 nick (0 ops, 0 voices, 1 normal)
```

Expected behavior:
* /names formatted output and nicklist need to show the appropriate prefix/prefixes.

The behavior is correct if I revert back to 48a3baedd5119798cbc0030eb4ea0ff661f58cbb.  This suggests that fadee506642ee80f42027816cc6cb758dcb68011 may have had unintended side effects, though I lack sufficient understanding of the background behind that commit to suggest a proper fix.

**Comment by RyanSquared** [comment]
The bug is that when a user receives a `JOIN` message from the IRCd, it's added to the list of nicks in the channel. The commit that causes this behavior makes it so that if a prefix is already set (which, the IRCd **should** send if it's not the default via a `MODE` message), WeeChat will discard the `NAMES` result for that user (which contains the "proper" prefix).

A fix for this could be if WeeChat checks if the prefixes are not the same - if they aren't, update the prefix.

**Comment by flashcode** [comment]
Thanks for reporting, this was introduced by the fix for issue #1019.
It is now fixed.
